packages/test-config/jest-unit.config.common.ts

```diff
@@ -2,6 +2,11 @@ import { Config } from '@jest/types';
 import path from 'path';
 
 const config: Partial<Config.InitialProjectOptions> = {
+    moduleNameMapper: {
+        // Strip `.js` from imports
+        // https://github.com/swc-project/jest/issues/64#issuecomment-1029753225
+        '^(\\.{1,2}/.*)\\.js$': '$1',
+    },
```

Comment:
Of course Jest doesn't know any of this, so we have to re-strip the .js extensions.